src/Algebra/Module/Construct/Nagata.agda

```diff
@@ -0,0 +1,170 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The non-commutative analogue of Nagata's construction
+-- of the "idealization of a module", defined here
+-- on R-R-*bi*modules M over a ring R.
+--
+-- Elements of R ⋉ M are pairs |R| × |M|
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Module.Construct.Nagata where
+
+open import Algebra.Bundles
```

Comment:
I remain a big fan of putting in as many `using` clauses as possible. Once you use 8+ things from something, sure, that gets excessive, but otherwise, it serves as helpful documentation as well as forward-proofing the code.

Reply:
Done, plus pruning out redundant imports altogether.